### DISCUSSION TITLE
Apache bookkeeper disk space reclaimation after consuming all messages from producer

### DISCUSSION THREAD

**Discussion by pradeepj2604** [apache/pulsar#20574] [Q&A]
Hi Pulsar Team,

There is a one scenario i wanted to achieve in apache pulsar.

Cluster size:
3 Nodes
Hardware Conf: Each with 4 core, 8 GB with 100 GB instance.
Each node has 1 zk, 1 bookie and 1 broker.
Apache pulsar version - 2.10.x
Using default configuration of all components.

I created a persistence topic named tenant\namespace\topic with namespace having.
I have set namespace retention to storage limit to 1 GB and time to 30 minutes.
Even after setting this disk size of bookkeeper is not being reclaimed to 0 size even after consuming all records.

I am ingesting 100,000 records with payload size of 25 KB. So total size should be 2.38 GB.
What we want to achieve is whenever we consume all records the storage size should be equal to 0, currently which we are unable to achieve this.

Can you help me with this case?

**Comment by tisonkun** [comment]
cc @hangc0276 could you help check this question?
